pkg/mount/mount.go

```diff
@@ -78,17 +80,25 @@ func RecursiveUnmount(target string) error {
 		}
 		logrus.Debugf("Trying to unmount %s", m.Mountpoint)
 		err = Unmount(m.Mountpoint)
-		if err != nil && i == len(mounts)-1 {
-			if mounted, err := Mounted(m.Mountpoint); err != nil || mounted {
-				return err
+		if err != nil {
+			// If the error is EINVAL either this whole package is wrong (invalid flags passed to unmount(2)) or this is
+			// not a mountpoint (which is ok in this case).
+			// Meanwhile calling `Mounted()` is very expensive.
+			if err == syscall.EINVAL {
+				continue
+			}
+			if i == len(mounts)-1 {
+				if mounted, e := Mounted(m.Mountpoint); e != nil || mounted {
+					return err
```

Comment:
We're returning the original error here, correct?

Reply:
Yes

Reply:
Alright; wasn't sure if we wanted the error returned by `Mounted()` as well

Reply:
Not really because the `Mounted` call is just a sanity check and the original error is the real interest. We also can't wrap two errors but the low-level system error is what we really want to propagate.